mixins/createGroupRouteMixin.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * A function to create a mixin to be used where there is an
+ * optional group param in the route.
+ *
+ * Designed to be used with the GroupSelect component.
+ *
+ * @param key used for the remember key and the route segment
+ * @returns a Vue mixin
+ */
+export default function createGroupRouteMixin(key) {
+  const rememberId = key
+  function routePath(id) {
+    return `/${key}/` + (id === 0 ? '' : id)
+  }
+  return {
+    computed: {
+      groupid: {
```

Comment:
Anything using this mixin automatically gets a readable/writable `groupid` field.